ecs-agent/logger/log.go

```diff
@@ -234,15 +236,59 @@ func setInstanceLevelDefault() string {
 	return DEFAULT_LOGLEVEL
 }
 
+// SetConfigLogFile sets the default output file of the logger.
+func SetConfigLogFile(logFile string) {
+	Config.lock.Lock()
+	defer Config.lock.Unlock()
+
+	Config.logfile = logFile
+}
+
+// SetConfigLogFormat sets the output format of the logger.
+// e.g. json, xml, etc.
+func SetConfigLogFormat(logFormat string) {
+	Config.lock.Lock()
+	defer Config.lock.Unlock()
+
+	os.Setenv(LOG_OUTPUT_FORMAT_ENV_VAR, logFormat)
+	Config.outputFormat = logFormat
+}
+
+// SetConfigMaxFileSizeMB sets the max file size of a log file
+// in Megabytes before the logger rotates to a new file.
+func SetConfigMaxFileSizeMB(maxSizeInMB float64) {
```

Comment:
The existing `SetLevel` function takes care of reloading the seelog config but the new setter functions do not. I understand that now there are multiple setter functions, a user might want to call the setter functions and then trigger a (re)load manually by calling `InitSeeLog` once but this breaks the current pattern that setter functions take care of reloading the configuration.

Reply:
Why not make the new setter functions call `reloadConfig`?

Reply:
Discussed offline that `reloadConfig` doesn't help with reloading log format.

Reply:
Synced offline, Some setters will require a reload of the custom formatters, which is not something `reloadConfig` does. If we are always calling `InitSeeLog` after each setter anyways, the access pattern can be abstracted to call `InitSeeLog` manually after modifying desired config attributes

Reply:
Let's update the descriptions of the new setter functions to say that `InitSeeLog` needs to be called for the change to take effect.

Reply:
Can be done in a new PR.